client/core/trade.go

```diff
@@ -2982,7 +3001,39 @@ func (t *trackedTrade) confirmRedemption(match *matchTracker) (bool, error) {
 			t.dc.log.Errorf("Failed to update match in db %v", err)
 		}
 		return false, errors.New("swap was already refunded by the counterparty")
-	} else if err != nil {
+
+	case errors.Is(err, asset.ErrTxRejected):
+		match.redemptionRejected = true
+		// If we're the maker, give up. They don't have the secret anyway.
+		if match.Side == order.Maker {
+			match.MetaData.Proof.SelfRevoked = true
```

Comment:
Why do they definitely not have the secret? If the secret was correct in the transaction but it was rejected for some other reason, then they could get it.

Reply:
Rejected transactions are stored on chain, they spend gas but don't modify any state. Secrets of lost transactions could be seen in the mempool.

Reply:
Oh yeah. Good catch.

Reply:
Yeah I can understand mempool. I guess we can get receipts for failed transactions?